### PR TITLE
Update maintaining-balance-for-open-source-maintainers.md

### PR DESCRIPTION
**Pull request by tochinet**
Corrected a few french typos, and translated the title

- [x] Have you followed the [contributing guidelines](https://github.com/github/opensource.guide/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you explained what your changes do, and why they add value to the Guides?

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

-----

### CODE REVIEW

**Comment by tochinet** [comment]
Apparently, the error in the check comes with some weird unrelated page [Getting error: Peer authentication failed for user "postgres", when trying to get pgsql working with rails](https://stackoverflow.com/questions/18664074/getting-error-peer-authentication-failed-for-user-postgres-when-trying-to-ge) that seems to be referenced too many times.

**Review by jmeridth — APPROVED** [review]

**Comment by dattd33** [comment]
Done

**Comment by sumansuhag** [comment]
As a data science researcher with an innovative mindset from Jhajjar, Haryana, I've always been driven to contribute meaningfully to society. Earthquake Early Warning (EEW) systems, with their evolution from basic regional networks to advanced AI-powered platforms, excite me.

By leveraging cutting-edge technology, historical lessons, and forward-thinking innovation, EEW systems can become a cornerstone of disaster resilience worldwide. I'd love to dedicate my life to contributing to this field, utilizing data science and AI to enhance EEW systems and save lives.

**Comment by sumansuhag** [comment]
P-wave detection using STA/LTA (classic seismology)

Rapid magnitude estimation using machine learning (Random Forest)

Shaking intensity prediction at a given location

Real-time alert simulation

import numpy as np
from obspy.signal.trigger import classic_sta_lta, trigger_onset
from sklearn.ensemble import RandomForestRegressor
import matplotlib.pyplot as plt

# --- 1. Simulate Seismic Data Stream with P-wave and S-wave ---
np.random.seed(42)
n_samples = 6000
data = np.random.normal(0, 0.5, n_samples)
# Inject synthetic P-wave and S-wave (earthquake starts at 2000)
p_wave = np.hamming(150) * 3
s_wave = np.hamming(300) * 10
data[2000:2150] += p_wave
data[2150:2450] += s_wave

# --- 2. STA/LTA P-wave Detection ---
sta = 30
lta = 300
threshold_on = 3.5
threshold_off = 1.0
sta_lta = classic_sta_lta(data, sta, lta)
triggers = trigger_onset(sta_lta, threshold_on, threshold_off)

# --- 3. Feature Extraction from First 3 Seconds after Trigger ---
fs = 100  # samples per second (simulate)
features = []
trigger_points = []
for onset, offset in triggers:
    if onset + 3*fs < len(data):
        segment = data[onset:onset+3*fs]
        feat = [segment.max(), segment.min(), segment.std(), np.abs(segment).mean()]
        features.append(feat)
        trigger_points.append(onset)

# --- 4. Rapid Magnitude Estimation (using ML) ---
# For demonstration, train a simple RandomForest on synthetic data
# In practice, use real labeled earthquake data!
X_train = np.random.rand(100, 4)
y_train = 4 + X_train[:,0]*2 + X_train[:,1] + np.random.normal(0, 0.2, 100)  # Fake magnitude
rf = RandomForestRegressor().fit(X_train, y_train)

# Predict magnitude for detected events
if features:
    features = np.array(features)
    mag_preds = rf.predict(features)
else:
    mag_preds = []

# --- 5. Shaking Intensity Prediction (Simplified) ---
def predict_intensity(mag, distance_km):
    # Simple ground motion prediction equation (GMPE)
    return mag - np.log10(distance_km + 1) * 1.2

# --- 6. Real-time Alert Simulation ---
for i, onset in enumerate(trigger_points):
    mag = mag_preds[i]
    distance_km = 30  # Assume user is 30 km from epicenter
    intensity = predict_intensity(mag, distance_km)
    print(f"ALERT: Earthquake detected at index {onset}. Estimated Magnitude: {mag:.2f}, "
          f"Predicted shaking intensity at {distance_km} km: {intensity:.2f}")

# --- 7. Visualization ---
plt.figure(figsize=(12, 4))
plt.plot(data, label='Seismic Data')
for onset in trigger_points:
    plt.axvline(onset, color='r', linestyle='--', alpha=0.7)
plt.title("Simulated Seismic Data with EEW Detection")
plt.legend()
plt.show()

**Comment by sumansuhag** [comment]
ensors:

Accelerometer (e.g., MPU6050)

Vibration sensor (e.g., SW-420)

Microcontroller:

ESP32 or Arduino (with Wi-Fi for IoT)

Automation/Alert:

Buzzer for local alert

Wi-Fi module for sending data/alerts to the cloud or mobile apps

Code:

#include <Wire.h>
#include <MPU6050.h>
#include <WiFi.h>

#define VIBRATION_PIN 4
#define BUZZER_PIN 5

MPU6050 mpu;
int vibrationValue = 0;

void setup() {
  Serial.begin(115200);
  pinMode(VIBRATION_PIN, INPUT);
  pinMode(BUZZER_PIN, OUTPUT);
  Wire.begin();
  mpu.initialize();
  // Wi-Fi setup code here
}

void loop() {
  vibrationValue = digitalRead(VIBRATION_PIN);
  int16_t ax, ay, az;
  mpu.getAcceleration(&ax, &ay, &az);

  // Simple threshold logic for earthquake detection
  if (vibrationValue == HIGH || abs(ax) > 15000 || abs(ay) > 15000 || abs(az) > 15000) {
    digitalWrite(BUZZER_PIN, HIGH); // Alert
    // Send alert to IoT server/cloud
    Serial.println("EARTHQUAKE DETECTED!");
    // Add Wi-Fi/MQTT/HTTP code to send data
  } else {
    digitalWrite(BUZZER_PIN, LOW);
  }
  delay(100);
}
